src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#ifndef IDYNTREE_ATTITUDEESTIMATOR_H
+#define IDYNTREE_ATTITUDEESTIMATOR_H
+
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Core/Rotation.h>
+
+#include <iostream>
+
+namespace iDynTree
+{
+    typedef iDynTree::Vector3 LinearAccelerometerMeasurements;
+    typedef iDynTree::Vector3 GyroscopeMeasurements;
+    typedef iDynTree::Vector3 MagnetometerMeasurements;
+
+    typedef iDynTree::Vector4 Quaternion;
+    typedef iDynTree::Vector3 RPY;
+
+    /**
+     * @class IAttitudeEstimator generic interface for attitude estimator classes
+     *
+     * The aim is to implement different attitude estimators as a block that takes IMU measurements
+     * as inputs and gives attitude estimates as outputs. This way the underlying implementation is abstracted
+     * and the user only has to set a few parameters and run the estimator.
+     *
+     * The general procedure to use the estimators would be,
+     * - instantiate the filter,
+     * - set initial internal state
+     * - in a loop,
+     *     - update the filter with measurements
+     *     - propagate the states
+     *
+     * However, additional methods to set and get parameters for the filter might be available with respect to the filters.
+     */
+    class IAttitudeEstimator
+    {
+    public:
+        virtual ~IAttitudeEstimator();
+
+        /**
+         * @brief Update the filter with accelerometer and gyroscope measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         * @return true/false if successful/not
```

Comment:
What happens if I call this method on a filter that uses the magnetometer info?

Reply:
I did not fully understand the question.

Reply:
What I meant is. In a filter where `m_params.use_magnetometer_measurements` is set to true, what happens if you call this method? Looking at the implementation it seems that it returns `true`, and then it continues to use the old magnetometer value.

Reply:
So you saying that, there needs to be ac cyclic call to the case where the magnetometer measurements are used and where they are not used ? Current implementation has only one-way possibility

Reply:
Sorry, I just meant that the behavior needs to be documented, not that the current behaviour needs to be changed.